src/main.lua

```diff
@@ -720,6 +721,58 @@ addEventingHandler(ActionMap.BuyRecord, utils.hasMatchingTag("Action", ActionMap
 	})
 end)
 
+addEventingHandler("upgradeName", utils.hasMatchingTag("Action", ActionMap.UpgradeName), function(msg)
+	-- get the name
+	local name = string.lower(msg.Tags.Name)
+	local from = utils.formatAddress(msg.From)
+	local timestamp = tonumber(msg.Timestamp)
+
+	if not timestamp then
+		ao.send({
+			Target = from,
+			Tags = { Action = "Invalid-" .. ActionMap.UpgradeName .. "-Notice", Error = "Bad-Input" },
+			Data = "Timestamp is required",
+		})
+		return
+	end
+
+	if not name then
+		ao.send({
+			Target = from,
+			Tags = { Action = "Invalid-" .. ActionMap.UpgradeName .. "-Notice", Error = "Bad-Input" },
+			Data = "Name is required",
+		})
+		return
+	end
+
+	local status, updatedRecordOrError = pcall(arns.upgradeRecord, from, name, timestamp)
```

Comment:
You can use eventingPcall here without consideration for other event data that might be relevant.

Reply:
[4730b49](https://github.com/ar-io/ar-io-network-process/pull/150/commits/4730b490caeb77ded5e39908df2e0b3223aeac88)